src/wfs/src/main/java/org/geoserver/wfs/response/CSVOutputFormat.java

```diff
@@ -269,7 +271,14 @@ private AttrFormatter[] getFormatters(FeatureType schema) {
                 } else if (java.sql.Time.class.isAssignableFrom(binding)) {
                     formatters[i] = sqlTimeFormatter;
                 } else if (java.util.Date.class.isAssignableFrom(binding)) {
-                    formatters[i] = juDateFormatter;
+                    if (gs.getService(WFSInfo.class) != null
+                            && gs.getService(WFSInfo.class).getCsvDateFormat() != null) {
+                        formatters[i] =
+                                new CustomDateFormatter(
+                                        gs.getService(WFSInfo.class).getCsvDateFormat());
+                    } else {
+                        formatters[i] = juDateFormatter;
+                    }
```

Comment:
Better, but still unnecessarily complicated, you're extracting wfsInto three times and the csv format two times...
Suggestion (not sure it compiles, but should give you an idea):

```
formatters[i] = Optional.ofNullable(gs.geService(WFSInfo.class))
                                        .map(wfs -> wfs.getCsvDateFormat())
                                        .map(format -> new CustomDateFormatter(format))
                                        .orElse(juDateFormatter)
```

Reply:
Done